raiden/network/transport/matrix/transport.py

```diff
@@ -777,6 +770,13 @@ def _handle_invite(self, room_id: _RoomID, state: dict) -> None:
 
         assert room is not None, f"joining room {room} failed"
 
+        if self._is_broadcast_room(room):
+            # This shouldn't happen with well behaving nodes but we need to defend against it
+            # Since we already are a member of all broadcast rooms, the `join()` above is in
+            # effect a no-op
+            self.log.warning("Got invite to broadcast room, ignoring", inviting_user=user)
```

Comment:
I think this should be an error level log. 

Though, i am worried that if this happens we'll never be able to know about it. Raising `RaidenUnrecoverableError` on the other hand can become an attack vector. 

Any idea how we can trigger a user to report a bug for this without crashing the node?

Reply:
Yeah I thought about that as well. I'm not sure it's `error` worthy since it's not an error in the node thats loggingg but rather someone else did something wrong.

Assuming we had a way to inform the user about this and they did open an issue, what should we do about it? Try to find out who is running the offending node?

Reply:
This basically tells us that a node tried to invite another into a broadcast room... but i see you're point. On the receiving end, there's not much to debug. What happened wrong in this case would be on inviting node.

Reply:
We can raise a recoverable error, as long as the error is properly handled when run on production mode.

Reply:
I don't think a raising a recoverable error will work here (at least not without an architecture change). 
`_handle_message()` is called from within the handle_thread greenlet (which is ultimately linked to the transport greenlet).

That means raising an exception there will skip handling of all remaining events from the last sync call and also cause the transport to re-raise the exception.

Reply:
@ulope True, I didn't think about that.

As a side note, not losing the message can be fixed by the processing queue PR https://github.com/raiden-network/raiden/pull/5275 if anyone wants to pick that one up (there is a race condition there that must be fixed, otherwise the exact problem you described still happens).